PlayniteSounds.cs

```diff
@@ -23,136 +21,253 @@
             // Add code to be executed when game is preparing to be started.
-            PlayFileName("GameStopped.wav");
+            PlaySoundFile(SoundFile.DesktopGameStoppedSound, SoundFile.FullscreenGameStoppedSound);
             ResumeMusic();
         }
 
         public override void OnGameUninstalled(OnGameUninstalledEventArgs args)
         {
             // Add code to be executed when game is uninstalled.
-            PlayFileName("GameUninstalled.wav");
+            PlaySoundFile(SoundFile.DesktopGameUninstalledSound, SoundFile.FullscreenGameUninstalledSound);
         }
 
         public override void OnApplicationStarted(OnApplicationStartedEventArgs args)
         {
             // Add code to be executed when Playnite is initialized.
-            PlayFileName("ApplicationStarted.wav");
-            SystemEvents.PowerModeChanged += OnPowerMode_Changed;
-            Application.Current.Deactivated += onApplicationDeactivate;
-            Application.Current.Activated += onApplicationActivate;
-            Application.Current.MainWindow.StateChanged += onWindowStateChanged;
+            PlaySoundFile(SoundFile.DesktopApplicationStartedSound, SoundFile.FullscreenApplicationStartedSound);
+            SystemEvents.PowerModeChanged += OnPowerModeChanged;
+            Application.Current.Deactivated += OnApplicationDeactivate;
+            Application.Current.Activated += OnApplicationActivate;
+            Application.Current.MainWindow.StateChanged += OnWindowStateChanged;
+        }
+
+        public override void OnApplicationStopped(OnApplicationStoppedEventArgs args)
+        {
+            // Add code to be executed when Playnite is shutting down.
+            SystemEvents.PowerModeChanged -= OnPowerModeChanged;
+            Application.Current.Deactivated -= OnApplicationDeactivate;
+            Application.Current.Activated -= OnApplicationActivate;
+            Application.Current.MainWindow.StateChanged -= OnWindowStateChanged;
+            PlaySoundFile(SoundFile.DesktopApplicationStoppedSound, SoundFile.FullscreenApplicationStoppedSound, true);
+            CloseAudioFiles();
+            CloseMusic();
+            _musicPlayer = null;
+        }
+
+        public override void OnLibraryUpdated(OnLibraryUpdatedEventArgs args)
+        {
+            // Add code to be executed when library is updated.
+            PlaySoundFile(SoundFile.DesktopLibraryUpdatedSound, SoundFile.FullscreenLibraryUpdatedSound);
+
+            if (Settings.AutoDownload)
+            {
+                var games = PlayniteApi.Database.Games.Where(x => x.Added != null && x.Added > Settings.LastAutoLibUpdateAssetsDownload);
+                CreateDownloadDialogue(games);
+            }
+
+            Settings.LastAutoLibUpdateAssetsDownload = DateTime.Now;
+            SavePluginSettings(Settings);
+        }
+
+        private void PlaySoundFile(string desktopSound, string fullscreenSound, bool useSoundPlay = false)
```

Comment:
I don't feel super great about this function. This was the best way to use the predefined constants for both modes, but it may just be better to assemble these on the fly, like it was done before.